### PR TITLE
[AIRFLOW-5667] Improve type annotations in GCP

### PR DESCRIPTION
**Pull request by TobKed**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5667
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: airflow/gcp/hooks/tasks.py [hunk 1]
```diff
@@ -505,14 +507,14 @@ def create_task(
...
         self,
-        location,
-        queue_name,
-        task_name,
-        project_id=None,
+        location: str,
+        queue_name: str,
+        task_name: str,
+        project_id: Optional[str] = None,
         response_view=None,
```
Comment (mik-laj) [inline airflow/gcp/hooks/tasks.py]: Why did you miss it?

File: airflow/gcp/hooks/tasks.py [hunk 2]
```diff
@@ -556,14 +558,14 @@ def get_task(
...
     def list_tasks(
         self,
-        location,
-        queue_name,
-        project_id=None,
+        location: str,
+        queue_name: str,
+        project_id: Optional[str] = None,
         response_view=None,
         page_size=None,
```
Comment (mik-laj) [inline airflow/gcp/hooks/tasks.py]: Can you add type annotation here also?

File: airflow/gcp/hooks/tasks.py [hunk 3]
```diff
@@ -68,14 +70,14 @@ def get_conn(self):
...
-        metadata=None,
-    ):
+        location: str,
+        task_queue: Union[Dict, Queue],
+        project_id: Optional[str] = None,
+        queue_name: Optional[str] = None,
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = None
+    ) -> Queue:
```
Comment (mik-laj) [inline airflow/gcp/hooks/tasks.py:80]: Can you add return type in other places?

File: airflow/gcp/hooks/tasks.py (hunk 1 above)
Comment (TobKed) [reply airflow/gcp/hooks/tasks.py]: I've fixed it :)

File: airflow/gcp/hooks/tasks.py (hunk 2 above)
Comment (TobKed) [reply airflow/gcp/hooks/tasks.py]: Sure, added.

File: airflow/gcp/hooks/tasks.py (hunk 3 above)
Comment (TobKed) [reply airflow/gcp/hooks/tasks.py:80]: Return types added as well.

File: airflow/gcp/hooks/cloud_sql.py [hunk 4]
```diff
@@ -924,17 +924,17 @@ def _generate_connection_uri(self) -> str:
...
-    def _get_sqlproxy_instance_specification(self):
+    def _get_sqlproxy_instance_specification(self) -> str:
         instance_specification = self._get_instance_socket_name()
         if self.sql_proxy_use_tcp:
             instance_specification += "=tcp:" + str(self.sql_proxy_tcp_port)
         return instance_specification
 
     @provide_session
-    def create_connection(self, session: Session = None):
+    def create_connection(self, session: Session = None) -> None:
```
Comment (simpleranchero) [inline airflow/gcp/hooks/cloud_sql.py]: Optional[Session]

Comment (TobKed) [reply airflow/gcp/hooks/cloud_sql.py]: Thanks! I've missed it. Fixed.

**Review by mik-laj — APPROVED** [review]

**Comment by turbaszek** [comment]
Hi @TobKed, do you think it's worth to add `Dict` type for `context` in `execute` methods?

**Comment by TobKed** [comment]
@nuclearpinguin I was thinking about it and I am not sure about it. We can discuss it offline and create separate PR if need.

**Review by mik-laj — APPROVED** [review]

**Pull request merged**